lecture_1/homework/app/models/warrior.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class Warrior < ApplicationRecord
+    belongs_to :clan
+    belongs_to :defensible, polymorphic: true
+    has_one :weapon
+
+    validates_presence_of :weapon
+  
+    validates :name, presence: true, length: { maximum: 256 }
+    validates_uniqueness_of :name, scope: :clan_id, conditions: -> { where(death_date: 'nil') }, message: "There can be only one living samurai of that name in a clan."
```

Comment:
Prawidłowa walidacja 👍 
Możesz jeszcze spróbować zapisać to w formie `validates :name, uniqueness: { ... }`, żeby zachować spójność z pozostałymi

Reply:
Racja!